### PR TITLE
Moved build_docs.py and BUILD into /tools/docs/

### LINKED ISSUE #2161: Refactor docs directory
Per comment https://github.com/tensorflow/addons/pull/2160#discussion_r487230647, better to move these stuff out of `docs`.

- [ ] Move build tools to `tools/docs`
- [ ] Modify related documentation

### PR DESCRIPTION
**Pull request by hp77-creator**
# Description

Brief Description of the PR:
To supplement the changes made in the location of files in docs/ to /tools/docs/

Fixes #2161 

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by hp77-creator** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Faddons%2Fpull%2F2167) for more info**.

@googlebot I signed it!

**Review by seanpmorgan — CHANGES_REQUESTED** [review]
Per discussion in #2161 please update the documentation in this PR

**Comment by hp77-creator** [comment]
> Per discussion in #2161 please update the documentation in this PR

Regarding that I had doubt, Shall I completely remove the documentation related with build_docs.py in docs/ and make a new documentation in tools/docs/ like it was in docs/ ?

**Comment by Harsh188** [comment]
Update [`docker/sanity_check.Dockerfile`](https://github.com/tensorflow/addons/blob/dbcd5aaef6da32bcd24040c55a49cef1538ef765/tools/docker/sanity_check.Dockerfile#L97) `RUN python docs/build_docs.py` to `RUN python tools/build_docs.py`. This should fix the build docs failure.

File: tools/docs/BUILD [hunk 1]
```diff
@@ -9,10 +9,3 @@ package(
     default_visibility = ["//tensorflow_addons:__subpackages__"],
 )
 
-py_binary(
-    name = "build_docs",
-    srcs = ["build_docs.py"],
-    deps = [
-        "//tensorflow_addons",
-    ],
-)
```
Comment (WindQAQ) [inline tools/docs/BUILD:18]: Revert this change.

**Review by WindQAQ — CHANGES_REQUESTED** [review]
Hi, thanks for contribution! We also have to modify this line
https://github.com/tensorflow/addons/blob/master/tools/docker/sanity_check.Dockerfile#L97 to
```docker
RUN python tools/docs/build_docs.py
```

File: tools/docs/BUILD (hunk 1 above)
Comment (hp77-creator) [reply tools/docs/BUILD:18]: on it. thanks!

Comment (hp77-creator) [reply tools/docs/BUILD]: @WindQAQ please review the PR

**Review by Harsh188 — APPROVED** [review]

**Review by WindQAQ — APPROVED** [review]
Thanks!

**Pull request merged**